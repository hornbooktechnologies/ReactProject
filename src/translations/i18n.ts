import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
