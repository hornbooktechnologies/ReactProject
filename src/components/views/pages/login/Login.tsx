import React from 'react'
import { useLoginState } from '../../../../hooks/useLoginState'
import i18n from '../../../../translations/i18n'

/**
 * Login with Microsoft Azure Ad
 *
 * On click of Login Button, open modal popup (redirect to url if it is internet explorer browser)
 * to authenticate user by microsoft account login. Once user authenticated, redirect back to
 * login page and execute further process of authentication.
 *
 * If Azure Client ID configured, Show Error message for client ID not configured and
 * Disable Login button
 *
 * Show error message if there is either client configuration error or API request fails
 */
const Login: React.FC = () => {
  const { handleLoginClick, isClientConfigured  } = useLoginState()

  return (
    <div className="main-content no-sidebar">
    {/* Content start */}
    <div id="content-section" className="content">
      <div className="h-100vh uk-flex uk-flex-middle uk-position-relative">
        <button
          className="uk-button uk-button-primary uk-button-large uk-margin-auto btn-shadow"
          onClick={() => handleLoginClick()}
          disabled={isClientConfigured ? false : true}
        >
          {i18n.t("LOGIN.LOGIN_BTN")}
        </button>
      </div>
    </div>
  </div>
  )
}

export default Login
