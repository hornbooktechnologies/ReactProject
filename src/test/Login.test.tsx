import React from "react";
import renderer from "react-test-renderer";
import Login from "../components/views/pages/login/Login";
import i18n from "../i18nForTests";
import { I18nextProvider } from "react-i18next";


describe("Component: Login", () => {
  it("Login renders correctly", () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <Login/>
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
