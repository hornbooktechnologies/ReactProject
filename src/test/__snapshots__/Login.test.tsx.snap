// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Login Login renders correctly 1`] = `
<div
  className="main-content no-sidebar"
>
  <div
    className="content"
    id="content-section"
  >
    <div
      className="h-100vh uk-flex uk-flex-middle uk-position-relative"
    >
      <button
        className="uk-button uk-button-primary uk-button-large uk-margin-auto btn-shadow"
        disabled={true}
        onClick={[Function]}
      >
        LOGIN.LOGIN_BTN
      </button>
    </div>
  </div>
</div>
`;
